proposals/1/standardized-mod-format.md

```diff
@@ -0,0 +1,126 @@
+# Standardized Mod Format (DRAFT!)
+
+## Description Of Issue
+
+Lack of a clear specification between modloaders on a common supported format.
+
+## Solution
+
+A specification which exactly defines specific terms, and does not define legacy formats from the previous era.
+
+### Advantages
+
+- Modloaders and modders know what is supported everywhere.
+
+- Less likelihood to use outdated mechanisms from the old era
+
+- APIs defined in other specifications may gain reference implementations using this specification - these can then become the only implementation of those APIs, preventing fragmentation.
+
+### Disadvantages
+
+- Per-modloader extensions not covered, theoretically limiting features for more portable mods.
+
+### Regarding Implementation
+
+It is important to note that some features of this specification may, themselves, be implemented within mods.
+
+This is fine - the implementation of this specification is then a matter of "that modloader + these mods".
+
+This in particular applies to how CCLoader handles patch files via `simplify`.
+
+## Mod Format
+
+A mod is in a directory. The directory's name is the ID of the mod.
+
+The directory then contains a file called "package.json".
+
+### The package.json format
+
+The package.json format describes a mod.
+
+It has the following elements:
+
+```
+{
+ "version": <Semver-compliant version string>,
+ "ccmodDependencies": <Optional Object - keys are mod IDs, values are semver-compliant version constraints - This is used between different mods>,
+ "dependencies": <
+  Optional Object - due to compatibility, this is a bit awkward
+  If ccmodDependencies is provided, keys are Node modules, values are semver-compliant version constraints
+  If ccmodDependencies is NOT provided, this key is effectively renamed to it - meaning keys are mod IDs, not Node modules.
+ >,
+
+ "name": <Optional human-readable string - otherwise, the directory name is used>,
```

Comment:
Should this be the human-readable string? The npm spec specifies strict limits on the package name.